pkg/controllers/pod_controller.go

```diff
@@ -102,6 +103,16 @@ func (r *PodReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.R
 	}
 
 	// logic for handling leader pod
+	// workerStsReplicas := *leaderWorkerSet.Spec.LeaderWorkerTemplate.Size - 1
+	if leaderWorkerSet.Spec.StartupPolicy == leaderworkerset.LeaderReadyStartupPolicy {
```

Comment:
Combine two if together with `&&`.

Reply:
No, we shouldn't. When leader pod is not ready if `spec.startpolicy=LeaderReady`, we should wait controller to reconcile next time